doc/specs/PtRunPlugin.md

```diff
@@ -0,0 +1,141 @@
+
+Lots of instances inside the settings dialog for how something like this could be enacted.  
+
+Here the options are both inline and allow for an on/off.
+
+![alt Windows 10 language inline options][WinSettingLanguage]
+
+![alt Windows 10 OS systray dialog][WinSettingSysIcon]
+
+### 1.4: Goals/Non-goals
+
+#### 1.4.1: Goals
+
+- Ability to customize action words
+- Ability to add a weight multiplier to increase weighting for individual plugins
+- Ability to disable a plugin
+
+#### 1.4.2: Stretch Goals
+
+- Ability to launch a single plugin via key shortcut
+- Have configurations happen without restarting PT Run
+
+#### 1.4.3: Non-goals
+
+- Settings outside what is listed above
+- 3rd party plugin spec
+
+## 2: Scenarios
+
+### 2.1: Golden path
+
+1. Disable a built-in plugin
+2. Change the weight of a plugin
+3. Change the action keyword of a plugin
+4. Trigger PT Run
+5. New configuration respected
+
+## 3: Key definitions
+
+| Term | Definition |
+|------|------------|
+| Action word | This is the concept inside PowerToys Run where you can directly invoke only that plugin(s) that use that.  An example is doing `=2+2` would only call the calculator plugin.  The UX Screenshot is calling this "Direct activation" below |
+| weight | A multiplier to increase the result value |
+| PT Run / Run | Shorthand for PowerToys Run launcher |
+
+## 4: Requirements
+
+### 4.1: Core requirements
+
+Most plugin information comes from `PowerToys\modules\launcher\Plugins\PLUGIN_FOLDER\plugin.json` where `PLUGIN_FOLDER` is the name of the plugin.
+
+| # | Definition | Priority |
+|---|------------|----------|
+| 1 | List of plugins with toggle On / Off | P0 |
+| 2 | When clicked, inline options are displayed | P0 |
+| 3 | When weight multiplier is added, that target plugin's value should be increased | P0 |
+| 4 | When new settings are applied, PT Run does not have to restart | P2 |
+| 5 | When an error state is hit, the warning will be displayed to user below title for plugin | P0 |
+| 6 | Detect error state when there is no direct activation, no global result and no action word | P0 |
+| 7 | Disable file drive warning checkbox if Search plugin is disabled  | P1 |
+| 8 | Do not have warning appear if Disable file drive warning is enabled and the Search plugin is disabled  | P1 |
+
+### 4.2: Inline Setting requirements
+
+| # | Definition | Priority |
+|---|------------|----------|
+| 1 | Checkbox to "include results in default list" (adding in * to action keyword)  | P0 |
+| 2 | Text field for additional action word. | P0 |
+| 3 | Action word do not have to be unique (This already is supported). | P0 |
+| 4 | Text field for additional action word can be blank. | P0 |
+| 5 | Do not allow action word of * as under the hood this is the global keyword | P1 |
+| 6 | Weight multiplier number box field, 0 to 10 with ability to add in a decimal.  This will need to be tweaked once the multiplier is added in after real world testing. | P1 |
```

Comment:
What is meant by `This will need to be tweak`? Default multipliers number for plugins? Range of multiplier number?

Reply:
For the range and small / large precision.  10 may just be too high or a small jump of .01 may not be enough.  Until we see how this interacts with true code, we're speaking in hypotheticals for ranges